pex/dist_metadata.py

```diff
@@ -97,14 +98,60 @@ def _parse_sdist_package_info(sdist_path):
     return None
 
 
+def find_dist_info_file(
+    project_name,  # type: str
+    version,  # type: str
+    filename,  # type: str
+    listing,  # type: Iterable[str]
```

Comment:
Would consuming an iterator allow for short-circuiting faster here? Or does that not matter because there aren't enough files?

Reply:
The latter.